lib/entity.js

```diff
@@ -71,6 +86,21 @@ Entity.prototype.merge = function merge (snapshot) {
   return this;
 };
 
+/**
+ * Merge a property onto the instance.
+ *
+ * Given a name and a value, mergeProperty checks if the name corresponds to a
+ * function or an object.
+ * - If not either, the value is simply applied to the
+ * instance.
+ * - If value is an object and the name corresponds to a value that is
+ * also an object in the instance, it is merged.
+ * [CONFIRM] - If the name corresponds to a function, the funcition is executed with
+ * value passed as the parameter.
+ *
+ * @param  {String} name   the name of the property being merged.
+ * @param  {Object} value  the value of the property being merged.
+ */
 Entity.prototype.mergeProperty = function mergeProperty (name, value) {
   if (mergeProperties.size &&
```

Comment:
Sure thing. mergeProperty is used by a repository (like https://github.com/mateodelnorte/sourced-repo-mongo) when it has retrieved an object from a datastore and is re-hydrating that object. If the property is a simple object like `{ some: 'property' }`, it can easily be deserialized and will appear on the object from the repo, as expected. But if the original object was defined by some function or class, and has behavior on it, you need to re-instantiate that object. 

mergeProperty is executed for each key/property name on the object returned from the data store. So, if you had a Car entity and it had a Wheel property that was defined by a Wheel class, you'd do something like: 

```
function Wheel (status) {
  this.status = status;
}

Wheel.prototype.go = function () {
  this.status = 'going';
}

function Car () {
  this.id = null;
  this.wheel = new Wheel(); // for instantiating our default wheel, when we first 'new' up a Car

  Entity.apply(this, arguments);
}

util.inherits(Car, Entity);

Entity.mergeProperty(Car, 'interests', function (obj) {
  this.wheel = new Wheel(); // for instantiating our wheel from saved values in a database
});
```